common/lib/xmodule/xmodule/tests/test_sequence.py

```diff
@@ -3,49 +3,50 @@
 """
 # pylint: disable=no-member
 from datetime import timedelta
-import ddt
 from django.utils.timezone import now
 from freezegun import freeze_time
 from mock import Mock
 from xmodule.tests import get_test_system
 from xmodule.tests.helpers import StubUserService
-from xmodule.tests.xml import XModuleXmlImportTest
-from xmodule.tests.xml import factories as xml
+from xmodule.tests.xml import factories as xml, XModuleXmlImportTest
 from xmodule.x_module import STUDENT_VIEW
 from xmodule.seq_module import SequenceModule
 
+TODAY = now()
+FUTURE_DATE = TODAY + timedelta(days=30)
+PAST_FUTURE_DATE = FUTURE_DATE + timedelta(days=7)
 
-@ddt.ddt
-class SequenceBlockTestCase(XModuleXmlImportTest):
+
+class SequenceBlockTestBase(XModuleXmlImportTest):
     """
-    Tests for the Sequence Module.
+    Base class for tests of Sequence Module.
     """
-    TODAY = now()
-    TOMORROW = TODAY + timedelta(days=1)
-    DAY_AFTER_TOMORROW = TOMORROW + timedelta(days=1)
+    SELF_PACED = False  # optionally overridden by child classes
 
-    @classmethod
-    def setUpClass(cls):
-        super(SequenceBlockTestCase, cls).setUpClass()
+    def setUp(self):
+        super(SequenceBlockTestBase, self).setUp()
 
-        course_xml = cls._set_up_course_xml()
-        cls.course = cls.process_xml(course_xml)
-        cls._set_up_module_system(cls.course)
+        course_xml = self._set_up_course_xml(self.SELF_PACED)
+        self.course = self.process_xml(course_xml)
+        self._set_up_module_system(self.course)
 
-        for chapter_index in range(len(cls.course.get_children())):
-            chapter = cls._set_up_block(cls.course, chapter_index)
-            setattr(cls, 'chapter_{}'.format(chapter_index + 1), chapter)
+        for chapter_index in range(len(self.course.get_children())):
+            chapter = self._set_up_block(self.course, chapter_index)
+            setattr(self, 'chapter_{}'.format(chapter_index + 1), chapter)
 
             for sequence_index in range(len(chapter.get_children())):
-                sequence = cls._set_up_block(chapter, sequence_index)
-                setattr(cls, 'sequence_{}_{}'.format(chapter_index + 1, sequence_index + 1), sequence)
+                sequence = self._set_up_block(chapter, sequence_index)
+                setattr(self, 'sequence_{}_{}'.format(chapter_index + 1, sequence_index + 1), sequence)
 
-    @classmethod
-    def _set_up_course_xml(cls):
+    @staticmethod
+    def _set_up_course_xml(self_paced=False):
         """
         Sets up and returns XML course structure.
         """
-        course = xml.CourseFactory.build()
+        if self_paced:
+            course = xml.CourseFactory.build(end=str(FUTURE_DATE), self_paced=str(True))
+        else:
+            course = xml.CourseFactory.build()
```

Comment:
Why set the `end` date only for self_paced?  The above 4 lines may be easier to understand if they simply become:
```
course = xml.CourseFactory.build(end=str(FUTURE_DATE), self_paced=str(self.SELF_PACED))
```